### PR TITLE
K8s 1.16

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

Adds the changes for enabling k8s-1.16 test lanes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Please see kubevirt/kubevirtci#177 also, which contains the required kubevirtci changes.

@rmohr I found out that changes in other yamls already were contained, so I figured I wouldn't need to make changes in manifests/testing templateable.

**Release note**:

```release-note
Add provider for k8s-1.16
```

### CODE REVIEW

**Comment by dhiller** [comment]
see kubevirt/project-infra#248 and kubevirt/kubevirtci#177

File: staging/src/kubevirt.io/client-go/go.mod [hunk 1]
```diff
@@ -40,3 +40,5 @@ replace (
 	k8s.io/client-go => k8s.io/client-go v0.0.0-20190228174230-b40b2a5939e4
 	kubevirt.io/containerized-data-importer => kubevirt.io/containerized-data-importer v1.10.6
 )
+
+go 1.13
```
Comment (phoracek) [inline staging/src/kubevirt.io/client-go/go.mod]: How is this related to the PR?

Comment (dhiller) [reply staging/src/kubevirt.io/client-go/go.mod]: Good find, thanks! Removed.

**Comment by rmohr** [comment]
/approve
/hold

Let's see in kubevirt/kubevirt if the tests run fine.

**Comment by dhiller** [comment]
> Let's see if the tests run fine first.

No, they don't. Will debug locally.

**Comment by rmohr** [comment]
At least the issue seems pretty limited. Probably related to the `apps/v1` think on the DaemonSet. I guess that in our tests, we are still trying to look up the DaemonSets via the `extensions/v1beta1` clientset.

**Comment by dhiller** [comment]
@rmohr yep, you were right, it was exactly that.

**Comment by dhiller** [comment]
Found similar code in `tests/utils.go`, curious why tests using that code still work...

**Comment by dhiller** [comment]
> Found similar code in `tests/utils.go`, curious why tests using that code still work...

Reason was that these branches were only triggered if a command line flag was set.

**Comment by dhiller** [comment]
Looks like we have some flakiness here, last run one test timed out that didn't before

```
[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:component]VMIlifecycle [rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:component]Creating a VirtualMachineInstance when name is longer than 63 characters [test_id:1625]should start it
```

Latest run (still running ATM) another test timed out:

```
Expose service on a VMI replica set
  /root/go/src/kubevirt.io/kubevirt/tests/expose_test.go:259
    Expose ClusterIP service
    /root/go/src/kubevirt.io/kubevirt/tests/expose_test.go:293
      [test_id:1537]Should create a ClusterIP service on VMRS and connect to it [It]
      /root/go/src/kubevirt.io/kubevirt/tests/expose_test.go:297
```

**Comment by dhiller** [comment]
Locally I had another failing test:

```
[Fail] [rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:component]Expose Expose UDP service on a VMI Expose NodePort UDP service [It] [test_id:1536]Should expose a NodePort service on a VMI and connect to it 
/root/go/src/kubevirt.io/kubevirt/tests/expose_test.go:253              
```

**Comment by dhiller** [comment]
Repeated test `[crit:high][vendor:cnv-qe@redhat.com][level:component]VMIlifecycle [rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:component]Creating a VirtualMachineInstance when name is longer than 63 characters [test_id:1625]should start it` 50 times, no failure.

**Comment by rmohr** [comment]
/hold

commit from kubevirtci still needs to be configured.

**Comment by dhiller** [comment]
/hold cancel

kubevirtci commit on master is now in.

**Comment by dhiller** [comment]
image

Hrm :(

**Comment by dhiller** [comment]
@rmohr have rebased on latest master, can you please have a look again? Thanks!
